spec/factories/chargeback_rate.rb

```diff
@@ -11,4 +11,48 @@
       chargeback_rate.chargeback_rate_details << FactoryGirl.create(:chargeback_rate_detail_memory_allocated_with_tiers)
     end
   end
+
+  factory :chargeback_rate_with_compute_details, :parent => :chargeback_rate do
+    ignore do
+      per_time 'hourly'
+    end
+
+    after(:create) do |chargeback_rate, evaluator|
+      chargeback_rate.chargeback_rate_details <<
+        FactoryGirl.create(:chargeback_rate_detail_cpu_used_with_tiers, :per_time => evaluator.per_time)
+      chargeback_rate.chargeback_rate_details <<
+        FactoryGirl.create(:chargeback_rate_detail_cpu_allocated_with_tiers, :per_time => evaluator.per_time)
+      chargeback_rate.chargeback_rate_details <<
+        FactoryGirl.create(:chargeback_rate_detail_cpu_cores_used_with_tiers, :per_time => evaluator.per_time)
+      chargeback_rate.chargeback_rate_details <<
+        FactoryGirl.create(:chargeback_rate_detail_disk_io_used_with_tiers, :per_time => evaluator.per_time)
+      chargeback_rate.chargeback_rate_details <<
+        FactoryGirl.create(:chargeback_rate_detail_fixed_compute_cost_1_with_tiers, :per_time => evaluator.per_time)
+      chargeback_rate.chargeback_rate_details <<
+        FactoryGirl.create(:chargeback_rate_detail_fixed_compute_cost_2_with_tiers, :per_time => evaluator.per_time)
+      chargeback_rate.chargeback_rate_details <<
+        FactoryGirl.create(:chargeback_rate_detail_memory_allocated_with_tiers, :per_time => evaluator.per_time)
+      chargeback_rate.chargeback_rate_details <<
+        FactoryGirl.create(:chargeback_rate_detail_memory_used_with_tiers, :per_time => evaluator.per_time)
+      chargeback_rate.chargeback_rate_details <<
+        FactoryGirl.create(:chargeback_rate_detail_net_io_used_with_tiers, :per_time => evaluator.per_time)
```

Comment:
Lots of dup'd lines for the same operation:

```ruby
%i(
  chargeback_rate_detail_cpu_used_with_tiers
  chargeback_rate_detail_cpu_allocated_with_tiers
  chargeback_rate_detail_cpu_cores_used_with_tiers
  chargeback_rate_detail_disk_io_used_with_tiers
  etc
).each do |factory_name|
  chargeback_rate.chargeback_rate_details << FactoryGirl.create(factory_name, :per_time => evaluator.per_time)
end
```

(However, I'm going to tell you to use traits in that factory, too)